app/models/enterprise.rb

```diff
@@ -199,7 +189,7 @@ def self.distinct_count
   end
 
   def activated?
-    confirmed_at.present? && sells != 'unspecified'
+    contact.confirmed? && sells != 'unspecified'
```

Comment:
So assuming that the `contact` user always receives the enterprise's emails, and defaults to the `owner`. This makes the `owner` a bit superfluous though. I like your 'receives_notifications' flag idea better.